src/main/protocol-impl/java/com/mysql/cj/protocol/a/authentication/AwsIamAuthenticationTokenHelper.java

```diff
@@ -85,17 +85,25 @@ private Region getRdsRegion() {
     // Check Hostname
     final Pattern auroraDnsPattern =
         Pattern.compile(
-            "(.+)\\.(proxy-|cluster-|cluster-ro-|cluster-custom-)?[a-zA-Z0-9]+\\.([a-zA-Z0-9\\-]+)\\.rds\\.amazonaws\\.com",
+            "(.+)\\.(proxy-|cluster-|cluster-ro-|cluster-custom-)?[a-zA-Z0-9]+\\.([a-zA-Z0-9\\-]+)\\.rds\\.amazonaws\\.com(\\.cn)?",
```

Comment:
i think there might be some confusion here (?)

the first regex was supposed to try to match non-CN endpoints

and then there was the CN-matcher for CN endpoints

do you want to merge them so one regex matches both?

I would suggest to put the regex string as a static variable and in the comments put the patterns that are valid

Reply:
Added `(\\.cn)?` to the end because i noticed that the pattern in RdsUtils.java has it, but it doesnt have it here. 
Personally, I'd prefer not to mix them. I'd rather keep it simple with two separate patterns, than one more complex one. 
I will put them in static vars are you suggested though.